File: java/ImprimeNumeros.java
/**
 * Programa Java imprime um determinado intervalo de números.
 *
 * Objetivos:
 *   - uso de repetições com o comando while;
 *   - uso dos operadores de pós incremento (+1) e decremento (-1);
 *
 * Módulo: Lógica de programação | Capítulo: XXX | Página: XXX
 * Author: Eder Magalhães
 */
public class ImprimeNumeros {

    public static void main(String args[]) {
        int numeroInicio = 1;
        int numeroFim = 10;

        System.out.println("Os numeros de "+numeroInicio+" até "+numeroFim+" (ascendente):");
        //logica: enquanto o nro inicial for menor ou igual ao nro final
        while (numeroInicio <= numeroFim) {
            System.out.println("\t "+numeroInicio++);
        }

        //volta o valor da variavel inicial
        numeroInicio = 1;

        System.out.println("\nOs numeros de "+numeroInicio+" até "+numeroFim+" (descendente):");
        while (numeroFim >= numeroInicio) {
            System.out.println("\t "+numeroFim--);
        }
    }

}

File: python/imprimeNumeros.py
# -*- coding: utf-8 -*-

# Programa Python imprime um determinado intervalo de números.
# Objetivos:
#   - uso de repetições com o comando while.
#
# Acesse e execute esse código Python em: http://pythonfiddle.com/imprime-numeros-ap/
#
# Módulo: Lógica de programação
# Author: Eder Magalhães

numeroInicio = 1;
numeroFim = 10;

print "Os numeros de ",numeroInicio," até ",numeroFim," (ascendente):";

# em ordem ascendente
while (numeroInicio <= numeroFim):
    print "\t ",numeroInicio;
    numeroInicio+=1;

numeroInicio = 1;

print "Os numeros de ",numeroInicio," até ",numeroFim," (decrescente):";

# em ordem decrescente
while (numeroFim >= numeroInicio) :
   print "\t ",numeroFim;
   numeroFim-=1;

